compiler/modules/nback/scaffold.bal

```diff
@@ -373,6 +375,27 @@ class Scaffold {
             return used;
         }
     }
+
+    function initNarrowRegs(llvm:Builder builder, bir:Label label) returns BuildError? {
+        foreach var reg in self.blockUninitRegs[label] {
+            check buildNarrowReg(builder, self, reg);
+        }
+    }
+
+    function scheduleNarrowRegInit(bir:Label label, bir:NarrowRegister reg) {
+        self.blockUninitRegs[label].push(reg);
+    }
+
+    function unscheduleNarrowRegInit(bir:Label label, bir:Register reg) {
```

Comment:
This is called in a loop. So we have a loop over a loop. Is this really the best data structure for this.